image/image.go

```diff
@@ -293,6 +300,28 @@ func MockTrusted(mockTrusted []asserts.Assertion) (restore func()) {
 	}
 }
 
+func makeKernelChannel(kernelTrack, defaultChannel string) (string, error) {
+	if strings.Count(kernelTrack, "/") != 0 {
+		return "", fmt.Errorf("cannot use kernel-track %q: must be a track name only", kernelTrack)
+	}
+	kch, err := snap.ParseChannel(kernelTrack, "")
+	if err != nil {
+		return "", fmt.Errorf("cannot use kernel-track %q: %v", kernelTrack, err)
+	}
+	if kch.Track == "" {
+		return "", fmt.Errorf("cannot use kernel-track %q: please specify a track", kernelTrack)
+	}
+
+	if defaultChannel != "" {
+		dch, err := snap.ParseChannel(defaultChannel, "")
+		if err != nil {
+			return "", fmt.Errorf("internal error: cannot parse channel %q", defaultChannel)
```

Comment:
is it an internal error? given that it can come from the user?  shouldn't we perhaps do the parsing earlier than here for it?

Reply:
Indeed, good point! I added code that does the checking for the user provided channel earlier (there was code for this before but I removed it again to keep things smaller and then did not update the error). I think (given how trivial this is) adding it to this PR makes sense.